FaunaDB.Client/Types/RefV.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using FaunaDB.Query;
+using Newtonsoft.Json;
+
+namespace FaunaDB.Types
+{
+    /// <summary>
+    /// A FaunaDB ref type.
+    /// <para>
+    /// See <see href="https://fauna.com/documentation/queries#values-special_types">FaunaDB Special Types</see>.
+    /// </para>
+    /// </summary>
+    public class RefV : ScalarValue<string>
+    {
+        /// <summary>
+        /// Create a Ref from a string, such as <c>new Ref("databases/prydain")</c>.
+        /// </summary>
+        /// <param name="value">Value.</param>
+        public RefV(string value) : base(value)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+        }
+
+        protected internal override void WriteJson(JsonWriter writer) =>
+            writer.WriteObject("@ref", Value);
+    }
+
+    public abstract class ObjectRef : RefV
```

Comment:
That abstraction was copied from core. The idea to copy the abstraction and use strong types instead of factory methods was to avoid mistakes that this model could cause. There are only a few combination of `id`, `class` and `database` attributes that are allowed by the api and also there's a distinction between class refs and database refs (if I understood correctly). I'm not against that model, to be honest my very first implementation was like that.